### PR TITLE
comment feature is completed

### PR DESCRIPTION
**Pull request by MFQ**

### CODE REVIEW

**Comment by MFQ** [comment]
@naveed-ahmad  please review this pull request and let me know

File: app/assets/javascripts/comments.coffee [hunk 1]
```diff
@@ -0,0 +1,25 @@
+window.App ||= {}
```
Comment (mmahalwy) [inline app/assets/javascripts/comments.coffee]: File not needed?

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class CommentsController < ApplicationController
+
+  def fetch_by_word
+    @comments = Comment.where(:word_id => params[:id]).order('created_at desc')
```
Comment (mmahalwy) [inline app/controllers/comments_controller.rb]: `word_id: params[:id]`

**Comment by MFQ** [comment]
@mmahalwy please review it again

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    @comments = Comment.where(word_id: params[:id]).order('created_at desc')
+  end
+
+  def create
+    @comment = Comment.new(comments_params)
+    @comment.save
+  end
+
+  protected
+  def comments_params
```
Comment (naveed-ahmad) [inline app/controllers/comments_controller.rb]: Please save the user with comment as well.

File: app/assets/stylesheets/comments.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Comments controller here.
```
Comment (naveed-ahmad) [inline app/assets/stylesheets/comments.scss]: empty file, 🗑 this

File: app/models/comment.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+class Comment < ApplicationRecord
```
Comment (naveed-ahmad) [inline app/models/comment.rb]: ActiveAdmin has comment model, although it is scoped so we won't have conflict. But still, I feel we should call this `ReviewComment` or `ProofReadComment`.

Plus, we'll have `ReviewComment/ProofReadComment` for other data points later on. So maybe add polymorphic relation here? 

`belongs_to :reviewable, polymorphic: true`

then in modal that need review, we'l have `has_many :review_comments, as :reviewable`

Comment (naveed-ahmad) [reply app/models/comment.rb]: Add reviewable(polymorphic) and user association to this model.

File: app/views/comments/_comment.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<div class="row" style="border-bottom: 1px solid black;">
+  <p style="padding-left: 5%;"> <%= comment.message %>
```
Comment (naveed-ahmad) [inline app/views/comments/_comment.html.erb]: no style in HTML, please.

File: app/views/comments/create.js.erb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+$("#comment_message").val('');
+$('.js-comments').prepend('<%= escape_javascript(render @comment ) %>');
```
Comment (naveed-ahmad) [inline app/views/comments/create.js.erb]: RJS is old, let's switch to Turbolink :)

File: db/migrate/20180420185143_create_comments.rb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+class CreateComments < ActiveRecord::Migration[5.1]
+  def change
+    create_table :comments do |t|
+      t.text :message
+      t.integer :word_id
+
+      t.timestamps
+    end
```
Comment (naveed-ahmad) [inline db/migrate/20180420185143_create_comments.rb]: Indexes for user and reviewabe column.

File: app/views/comments/create.js.erb (hunk 7 above)
Comment (MFQ) [reply app/views/comments/create.js.erb]: @naveed-ahmad  How can I do that? as per my understanding, they have to remove partial replacement.

**Comment by naveed-ahmad** [comment]
@MFQ polymorphic PR :) 

https://github.com/quran/community.quran.com/pull/37

**Pull request closed without merging**